### PR TITLE
Clean util (resolves #298, resolves #313)

### LINKED ISSUE #298: Sort test leaves S3 buckets behind

### LINKED ISSUE #313: Create Toil clean util

### PR DESCRIPTION
**Pull request by cket**
I left the default value of the clean flag as 'never', I think it is safer to let people delete the jobStore themselves with the util. I implemented clean as part of the teardown in sort test, and it is tested itself in the utils unittest.

### CODE REVIEW

**Comment by hannes-ucsc** [comment]
Let's discuss your decision to have `--clean` default to `never` in tomorrows scrum.

**Comment by hannes-ucsc** [comment]
The conflicts are in sortTest which I recently streamlined. I think I would be the best person to resolve them. Ok, to go multi-author?

**Comment by cket** [comment]
Yeah, fine by me!

File: src/toil/utils/toilClean.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import logging
+import os
+import sys
+
+from toil.lib.bioio import getBasicOptionParser
+from toil.lib.bioio import parseBasicOptions
+from toil.common import loadJobStore
+from toil.jobStores.abstractJobStore import JobStoreCreationException
+
+logger = logging.getLogger( __name__ )
```
Comment (hannes-ucsc) [inline src/toil/utils/toilClean.py]: Missing a doc string? See if `toil --help` still works?

File: src/toil/utils/toilClean.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+def main():
+    """Reports the state of the toil.
```
Comment (hannes-ucsc) [inline src/toil/utils/toilClean.py]: Docstring is inaccurate.

File: src/toil/utils/toilClean.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    parser = getBasicOptionParser("usage: %prog [--jobStore] JOB_STORE", "%prog 0.1")
+
+    parser.add_option("--jobStore", dest="jobStore",
+                      help="Job store path. Can also be specified as the single argument to the script.\
+                       default=%default", default=os.path.abspath("./toil"))
+
+    options, args = parseBasicOptions(parser)
+    logger.info("Parsed arguments")
+
+    assert len(args) <= 1 #Only toil may be specified as argument
```
Comment (hannes-ucsc) [inline src/toil/utils/toilClean.py]: I don't understand this. The jobStore can be passed in via the "--jobStore" option and as a positional argument?

**Comment by hannes-ucsc** [comment]
Don't like the ambiguity between the positional and the option. What if a confused user specifies both?

Follow up in #329, #328 and #327

**Pull request merged**